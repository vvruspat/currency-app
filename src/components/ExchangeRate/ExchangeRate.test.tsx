import { render, screen } from "@testing-library/react";
import { ExchangeRate } from ".";

test("renders Exchange USD -> EUR", () => {
  render(
    <ExchangeRate
      data-testid="test-exchange"
      from="USD"
      to="EUR"
      rate={0.985676}
    />,
    {}
  );
  const ExchangeElement = screen.getByTestId("test-exchange");

  expect(ExchangeElement).toBeInTheDocument();
  expect(ExchangeElement).toMatchSnapshot();
});

test("renders Exchange EUR -> USD", () => {
  render(
    <ExchangeRate
      data-testid="test-exchange"
      from="EUR"
      to="USD"
      rate={1.985676}
    />,
    {}
  );
  const ExchangeElement = screen.getByTestId("test-exchange");
  expect(ExchangeElement).toBeInTheDocument();
  expect(ExchangeElement).toMatchSnapshot();
});
