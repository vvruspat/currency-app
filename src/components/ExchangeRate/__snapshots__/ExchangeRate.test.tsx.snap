// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Exchange EUR -> USD 1`] = `
<div
  class="exchange-rate"
  data-testid="test-exchange"
>
  <svg
    class="exchange-rate-chart-icon"
  >
    Chart16Icon.svg
  </svg>
  €1.00
   
  =
   
  $1.98568
</div>
`;

exports[`renders Exchange USD -> EUR 1`] = `
<div
  class="exchange-rate"
  data-testid="test-exchange"
>
  <svg
    class="exchange-rate-chart-icon"
  >
    Chart16Icon.svg
  </svg>
  $1.00
   
  =
   
  €0.98568
</div>
`;
